### PR TITLE
TS-3607: Integrate ats_pagespeed into the automake build

### PR DESCRIPTION
**Pull request by caguado**
This commit adds _STDC_LIMIT_MACROS and __STDC_CONSTANT_MACROS to
fix the missing macros defined in stdint.h when building a C++ source
code.

### CODE REVIEW

**Comment by oschaaf** [comment]
Great! LGTM

**Comment by jpeach** [comment]
Wait, why isn't page speed integrated into the automake build? We add these macros to the build globally at configuration time.

**Comment by oschaaf** [comment]
@jpeach Because the pagespeed libraries currently do not support all the platforms that ATS does, so some work has to be done to exclude the plugin in some platform builds.

**Comment by jpeach** [comment]
That's fine, we have other stuff that only builds when it's dependencies are available. I'm happy to advise if you want help doing this in autotools.

**Comment by caguado** [comment]
How does it look? I'm quite new in writing autotools. Is it ok to rename gzip.so to ats_pagespeed_gzip.so? Documentation would need an update too.

**Comment by oschaaf** [comment]
@caguado my 2p is that renaming the specific gzip module for ats_pagespeed is fine for now. 
From the top of my head the gzip version from ats_pagespeed is nearly identical to the real one. 
I will have a look at merging and eliminating this gzip plugin for later.

**Comment by bryancall** [comment]
https://issues.apache.org/jira/browse/TS-3607

**Comment by bryancall** [comment]
@oschaaf can you please work on this and get it committed?

**Comment by oschaaf** [comment]
@bryancall yes. We've got some changes in our pipeline as well, I'll follow up shortly.

**Comment by zwoop** [comment]
@oschaaf  Ping!

**Comment by oschaaf** [comment]
Change looks good to me! What is the flow with regard to committing a pull request?

**Comment by zwoop** [comment]
Easiest is to look at the Jira, which has instructions how to merge and push. Make sure you "git pull --rebase" before you push upstream to ASF, otherwise the "Committer" field isn't update properly. Basically, we want the author to be the person submitting the PR, but you as the committer should show up as well (so we can see who actually pushed it).

**Comment by oschaaf** [comment]
@zwoop thanks!

File: plugins/experimental/ats_pagespeed/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+else
+  os_name = unknown_os
+endif
+
+if ATS_PAGESPEED_X86_64
+  arch_name = x64
+else
+  arch_name = ia32
+endif
+
```
Comment (jpeach) [inline plugins/experimental/ats_pagespeed/Makefile.am]: You don't need to do this. In ``configure.ac`` changes, just ``AC_SUBST`` the values. Then later on you can use ``@ATS_PAGESPEED_OS@`` and ``@ATS_PAGESPEED_ARCH@``.

**Comment by zwoop** [comment]
Ping @oschaaf . Also, Otto, you must do the git-dual setup steps to make your account properly enabled here, otherwise, we can't assign PR's to you.

**Comment by oschaaf** [comment]
@zwoop done

**Comment by atsci** [comment]
Can one of the admins verify this patch?

**Comment by oschaaf** [comment]
The patch looks good to me, but @jpeach had an improvement in mind. @jpeach would you be OK with committing this and doing https://github.com/apache/trafficserver/pull/210#discussion_r57108812 in a follow-up?

**Comment by jpeach** [comment]
Sure either way is fine.

**Comment by atsci** [comment]
Can one of the admins verify this patch? Only approve PRs which have been reviewed.

**Comment by atsci** [comment]
Can one of the admins verify this patch? Only approve PRs which have been reviewed.

**Comment by atsci** [comment]
Can one of the admins verify this patch? Only approve PRs which have been reviewed.

**Comment by atsci** [comment]
Can one of the admins verify this patch? Only approve PRs which have been reviewed.

**Comment by atsci** [comment]
Can one of the admins verify this patch? Only approve PRs which have been reviewed.

**Comment by zwoop** [comment]
This unfortunately now runs into issues with clang-format. Can you rebase, and run clang-format again, and then force push? See

https://ci.trafficserver.apache.org/view/github/job/Github-Linux/70/

**Comment by zwoop** [comment]
.

**Comment by caguado** [comment]
Rebased against current 5.3.x and passed clang-format.

@jpeach, please let me know whether this is what you had in mind.

**Comment by zwoop** [comment]
We still need an update to this. @oschaaf and @jpeach , assigning to you for final reviews, but we should try to land this assuming it seems reasonable (once clang-format etc. has been addressed).

**Comment by bryancall** [comment]
This would be a great to add pagespeed to the build.  The patch would need to be updated against the master branch.

**Comment by bryancall** [comment]
I haven't seen an update to this PR for awhile.  I will close it in a week if it hasn't been updated.

**Comment by bryancall** [comment]
Please reopen or create another pull request if you want to resolve this.

**Pull request closed without merging**